### ISSUE TITLE
Support standardized configuration files

### ISSUE THREAD

**Issue by afeld** [Snowflake-Labs/schemachange#281]
**Is your feature request related to a problem? Please describe.**

My team uses a bunch of tools that connect to Snowflake. Unfortunately, their configuration file support is a bit all over the place:

| Tool                   | Supports `schemachange-config.yml` | Supports `config.toml` | Supports `connections.toml` |
| ---------------------- | ---------------------------------- | ---------------------- | --------------------------- |
| Schemachange           | [Yes][sc]                          | ?                      | [Yes][sc-connections]\*                           |
| Snowflake CLI          |                                    | [Yes][snow-cli-config]        | [Yes][snow-cli-connections]             |
| The Python Connector   |                                    | [Yes][python-config]   | [Yes][python-connections]   |
| The VSCode extension   |                                    |                        | [Yes][vscode]               |
| The Terraform provider |                                    | [Yes][tf]\*\*            |

[sc]: https://github.com/Snowflake-Labs/schemachange?tab=readme-ov-file#yaml-config-file
[sc-connections]: https://github.com/Snowflake-Labs/schemachange?tab=readme-ov-file#connectionstoml-file
[snow-cli-config]: https://docs.snowflake.com/en/developer-guide/snowflake-cli/connecting/configure-cli
[snow-cli-connections]: https://docs.snowflake.com/en/developer-guide/snowflake-cli/connecting/configure-connections#alternative-configuration-file
[python-config]: https://docs.snowflake.com/en/developer-guide/python-connector/python-connector-connect#setting-a-default-connection
[python-connections]: https://docs.snowflake.com/en/developer-guide/python-connector/python-connector-connect#connecting-using-the-connections-toml-file
[vscode]: https://docs.snowflake.com/en/user-guide/vscode-ext#edit-the-snowflake-connections-toml-file
[tf]: https://registry.terraform.io/providers/snowflake-labs/snowflake/latest/docs#config-file

\*[Coming in v4.0.0.](https://github.com/Snowflake-Labs/schemachange/blob/master/CHANGELOG.md#400---tbd)

\*\*It says "The Snowflake config file lives at `~/.snowflake/config` and uses TOML format" - I haven't tested if `~/.snowflake/config.toml` works as well.

This means that the configuration for each needs to be done separately, which was a hassle. (Less so now that it's all set up.)

**Describe the solution you'd like**

My ideal is that Snowflake as a company moves all their projects to standardized configuration files - presumably `config.toml` and `connections.toml` - with a canonical specification/documentation. I'm not sure where best to raise that idea so that the various project maintainers would see it.

Given this is the schemachange repository, my specific request is for schemachange to support those files. Given it uses the Python Connector under the hood, I suspect they might be supported already, but it's unclear to me how that plays with the `schemachange-config.yml` without digging through the code / doing a bunch of experimentation. If that's the case, I suppose the resolution would be an update to the README about how to use them.

**Describe alternatives you've considered**

Our workaround has been to specify the user-specific/sensitive configuration details in a `.env` file, then pass it through as environment variables by running our various commands (`schemachange`, `terraform`, etc.) through [dotenv](https://pypi.org/project/python-dotenv/).

**Additional context**

none

---

Thanks!

**Comment by zanebclark** [comment]
Agreed! I've closed https://github.com/Snowflake-Labs/schemachange/issues/291 in favor of your superior issue.

Check out https://github.com/Snowflake-Labs/schemachange/pull/298 for a partial implementation. Some inconsistencies you might be interested in: 
- The snowflake cli only supports a config.toml and the Snowflake Python connector only supports a connections.toml. There's a real lack of parity across the tools here.
- Some of the connections.toml arguments use hyphens and some use underscores.

**Comment by afeld** [comment]
> The snowflake cli only supports a config.toml

That's changed! Updated the table above.

**Comment by sfc-gh-tmathew** [comment]
@afeld With the release of v4.2.0, we are moving towards supporting snowflake connection parameters supported by the snowflake python connector. I don't see anything specifically to implement but that your idea has been incorporated in the switch to 4.0.0 and stablized as of 4.2.0 release. Let us know if the latest release aligns with your ask adn close the issue accordingly.
